app/src/main/java/org/simple/clinic/newentry/PatientEntryScreenController.kt

```diff
@@ -175,39 +144,16 @@ class PatientEntryScreenController @Inject constructor(
 
     val personDetailChanges = Observables.combineLatest(nameChanges, dateOfBirthChanges, ageChanges, genderChanges)
     { name, dateOfBirth, age, gender ->
-      OngoingPatientEntry.PersonalDetails(name, dateOfBirth.nullIfBlank(), age.nullIfBlank(), gender.toNullable())
+      PersonalDetails(name, dateOfBirth.nullIfBlank(), age.nullIfBlank(), gender.toNullable())
     }
 
-    val phoneNumberChanges = Observables
-        .combineLatest(
-            events
-                .ofType<PatientNoPhoneNumberToggled>()
-                .map { it.noneSelected },
-            events
-                .ofType<PatientPhoneNumberTextChanged>()
-                .map { it.phoneNumber })
-        .map { (noneSelected, phoneNumber) ->
-          when {
-            noneSelected -> None
-            phoneNumber.isBlank() -> None
-            else -> Just(OngoingPatientEntry.PhoneNumber(phoneNumber!!))
-          }
-        }
+    val phoneNumberChanges = events
+        .ofType<PatientPhoneNumberTextChanged>()
+        .map { (phoneNumber) -> if (phoneNumber.isBlank()) None else Just(PhoneNumber(phoneNumber)) }
```

Comment:
Is `Optional` needed here anymore? `colonyOrVillageChanges` doesn't use it.

Reply:
`colonyOrVillageChanges` is no longer an `Optional`. Did you mean `phoneNumberChanges`? It needs to be optional since phone number is still optional to enter.

Reply:
Ah okay

Reply:
Wait, you're not making use of `Optional` anywhere in this function, right? It gets immediately converted back to a nullable string using `toNullable`.